scripts-available/CDB_Overviews.sql

```diff
@@ -559,16 +577,15 @@ BEGIN
     ELSE
       RETURN Format('SUM(%s*%s)/%s::' || column_type, qualified_column, feature_count, total_feature_count);
     END IF;
-  WHEN 'text' THEN
-    -- TODO: we could define a new aggregate function that returns distinct
-    -- separated values with a limit, adding ellipsis if more values existed
-    -- e.g. with '/' as separator and a limit of three:
-    --     'A', 'B', 'A', 'C', 'D' => 'A/B/C/...'
-    -- Other ideas: if value is unique then use it, otherwise use something
-    -- like '*' or '(varies)' or '(multiple values)', or NULL
-    -- Using 'string_agg(' || qualified_column || ',''/'')'
-    -- here causes
-    RETURN 'CASE count(*) WHEN 1 THEN MIN(' || qualified_column || ') ELSE NULL END::' || column_type;
+  WHEN 'text', 'character varying', 'character' THEN
+    SELECT _cdb_unlimited_text_column(reloid, column_name) INTO unlimited_text;
+    IF unlimited_text THEN
+      -- TODO: this should not be applied to columns containing largish text;
+      -- it is intended only to short names/identifiers
+      RETURN  'CASE WHEN count(*) = 1 THEN MIN(' || qualified_column || ') WHEN ' || total_feature_count || ' < 5 THEN string_agg(distinct ' || qualified_column || ','' / '') ELSE ''*'' END::' || column_type;
```

Comment:
I have to check if that works and is efficient, because it would be a great improvement.
(I fear it may affect performance, specially for multiple text columns, because it has to count distinct values of each text column per group, instead of just using the group size).